include/dem/load_balancing.h

```diff
@@ -0,0 +1,398 @@
+/* ---------------------------------------------------------------------
+ *
+ * Copyright (C) 2019 - 2024 by the Lethe authors
+ *
+ * This file is part of the Lethe library
+ *
+ * The Lethe library is free software; you can use it, redistribute
+ * it, and/or modify it under the terms of the GNU Lesser General
+ * Public License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * The full text of the license can be found in the file LICENSE at
+ * the top level of the Lethe distribution.
+ *
+ * ---------------------------------------------------------------------
+ *
+ */
+
+
+#ifndef lethe_load_balancing_h
+#define lethe_load_balancing_h
+
+#include <core/parameters_lagrangian.h>
+#include <core/simulation_control.h>
+
+#include <dem/adaptive_sparse_contacts.h>
+#include <dem/data_containers.h>
+
+#include <deal.II/distributed/tria.h>
+
+#include <deal.II/particles/particle_handler.h>
+
+using namespace dealii;
+
+/**
+ * @brief Manages the load balancing (repartitioning), of the domain according
+ * to the computational load of the cells and the particles.
+ * It is used by the DEM and the coupling CFD-DEM solvers.
+ */
+template <int dim>
+class LoadBalancing
```

Comment:
I would maybe call the class LagrangianLoadBalancing or ParticlesLoadBalancing? Because eventually we could be doing load balancing  using something that does not have particles.